### DISCUSSION TITLE
Resuming play/stop position

### DISCUSSION THREAD

**Discussion by chz8494** [CoderLine/alphaTab#1084] [Q&A] [answered]
I was following your tutorial example and used clicked method, it works perfectly and can play again from the stopped position.
```javascript
          playPause.onclick = (e) => {
            if (e.target.classList.contains("disabled")) {
              return;
            }
            api.playPause();
          };
          stop.onclick = (e) => {
            if (e.target.classList.contains("disabled")) {
              return;
            }
            api.stop();
          };
``` 
but if I don't use this method, instead just call `api.stop()` directly from a socket.io function, it fails to remember the stop/pause position and keeps restarting from the very beginning. 
```javascript
          socket.on('play', function(){
            api.playPause();
          });
          socket.on('stop', function(){
            api.stop();
          });
```
would like to know what I'm missing and how to tell player where to continue.
Thanks.

**Comment by Danielku15** [comment, marked as answer]
The behavior is intended like you describe it. 

> Stops the playback of the current song, and moves the playback position back to the start. https://alphatab.net/docs/reference/api/stop/

Stop is like the stop button in a music player where playback is fully reset. The position is remembered on **pause**. 

https://alphatab.net/docs/reference/api/playpause/
https://alphatab.net/docs/reference/api/play/
https://alphatab.net/docs/reference/api/pause/

**Comment by chz8494** [comment]
Wow, thank you, I misread the stop function and didn't notice there's a pause.
